src/main/scala/no/gdl/bookapi/service/translation/TranslationService.scala

```diff
@@ -8,17 +8,103 @@
 package no.gdl.bookapi.service.translation
 
 import com.typesafe.scalalogging.LazyLogging
-import no.gdl.bookapi.model.api.{TranslateRequest, TranslateResponse}
+import io.digitallibrary.language.model.LanguageTag
+import no.gdl.bookapi.integration.crowdin.{CrowdinClient, CrowdinClientBuilder, CrowdinUtils}
+import no.gdl.bookapi.model._
+import no.gdl.bookapi.model.api._
+import no.gdl.bookapi.model.domain.{InTranslation, InTranslationFile, TranslationStatus}
+import no.gdl.bookapi.service.ReadService
 
-import scala.util.{Success, Try}
+import scala.util.{Failure, Success, Try}
 
 trait TranslationService {
+  this: CrowdinClientBuilder with ReadService with SupportedLanguageService with TranslationDbService =>
   val translationService: TranslationService
 
+
   class TranslationService extends LazyLogging {
-    def addTranslation(translateRequest: TranslateRequest): Try[TranslateResponse] = {
-      // TODO in #155: Add implementation of sending to crowdin
-      Success(TranslateResponse(1, "https://www.digitallibrary.io"))
+    def fetchTranslationsIfAllTranslated(projectIdentifier: String, language: LanguageTag): Try[Unit] = {
+      Success()
+      // TODO in #138: Implement fetch translations
+    }
+
+    def updateTranslationStatus(projectIdentifier: String, language: LanguageTag, fileId: String, status: TranslationStatus.Value): Try[InTranslationFile] = {
+      translationDbService.fileForCrowdinProjectWithFileIdAndLanguage(projectIdentifier, fileId, language) match {
+        case None => Failure(new NotFoundException(s"No translation for project $projectIdentifier, language $language and file_id $fileId"))
```

Comment:
Remember to fix `file_id` if `_` is omitted or replaced by`-`.

Reply:
Will not be replaced